### PR TITLE
1690 Molecularize CIF

### LINKED ISSUE #1690: Molecularize CIF handling a bit better
**Describe the feature / issue**
Data management within the `CIFHandler` could be tidied significantly if we introduce a sort of `StaticMolecule` class - one which inherits `Molecule` but which maintains and references its own `Atom` vector.  That way we can harmonise storage of data, make use of functions within the `Molecule` class (principally `unFold()`) and reduce code.

### PR DESCRIPTION
**Pull request by trisyoungs**
This PR focuses on our CIF import functionality, refactoring a few things here and there and generally making life a bit easier.  A bug is fixed in the generation of the supercell whereby individual `Species` were generated for every molecule instance, and the handling of various output options in the wizard was kinda broken.

It ended up being a bit larger than intended!

Closes #1690.

### CODE REVIEW

File: src/io/import/cif.h [hunk 1]
```diff
@@ -140,7 +140,7 @@ class CIFHandler
 
     private:
     // Create basic unit cell
-    bool createBasicUnitCel();
+    bool createBasicUnitCell();
```
Comment (jws-1) [inline src/io/import/cif.h:143]: Oops! Not sure how that survived here for so long!

File: src/io/import/cif.cpp [hunk 2]
```diff
@@ -577,61 +577,86 @@ bool CIFHandler::detectMolecules()
...
+        sp->setName(EmpiricalFormula::formula(sp->atoms(), [&](const auto &at) { return at.Z(); }));
 
-        // Find copies of this fragment
-        auto copies = speciesCopies(cleanedUnitCellSpecies_, neta.value());
+        // Fix geometry of the extracted species
+        fixGeometry(sp, cleanedUnitCellSpecies_->box());
 
-        // Remove the current fragment and all copies
-        for (auto &copy : copies)
+        // Find instances of this fragmentIndices. For large fragments that represent > 50% of the remaining atoms we don't even
```
Comment (jws-1) [inline src/io/import/cif.cpp]:
```suggestion
        // Find instances of this fragment. For large fragments that represent > 50% of the remaining atoms we don't even
```

File: src/io/import/cif.cpp [hunk 3]
```diff
@@ -577,61 +577,86 @@ bool CIFHandler::detectMolecules()
...
+        // attempt to create a NETA definition etc. For cases such as framework species this will speed up detection no end.
+        std::vector<CIFLocalMolecule> instances;
+        if (fragmentIndices.size() * 2 > allAtomIndices.size())
```
Comment (jws-1) [inline src/io/import/cif.cpp:612]: :+1:

File: src/io/import/cif.cpp [hunk 4]
```diff
@@ -577,61 +577,86 @@ bool CIFHandler::detectMolecules()
...
-                                         [&](int value) { return std::find(copy.begin(), copy.end(), value) != copy.end(); }),
-                          indices.end());
-        }
-
-        // Determine coordinates
-        auto coords = speciesCopiesCoordinatesFromUnitCell(sp, cleanedUnitCellSpecies_->box(), copies);
-
-        // Fix geometry
-        fixGeometry(sp, cleanedUnitCellSpecies_->box());
+            // Create an instance of the current fragmentIndices
```
Comment (jws-1) [inline src/io/import/cif.cpp]:
```suggestion
            // Create an instance of the current fragment
```

File: src/io/import/cif.cpp [hunk 5]
```diff
@@ -577,61 +577,86 @@ bool CIFHandler::detectMolecules()
...
+            allAtomIndices.erase(std::remove_if(allAtomIndices.begin(), allAtomIndices.end(),
+                                                [&](int value) {
+                                                    return std::find(fragmentIndices.begin(), fragmentIndices.end(), value) !=
+                                                           fragmentIndices.end();
+                                                }),
+                                 allAtomIndices.end());
+        }
+        else
+        {
+            // Determine a unique NETA definition describing the fragmentIndices
```
Comment (jws-1) [inline src/io/import/cif.cpp]:
```suggestion
            // Determine a unique NETA definition describing the fragment
```

File: src/io/import/cif.cpp [hunk 6]
```diff
@@ -577,61 +577,86 @@ bool CIFHandler::detectMolecules()
...
+            auto neta = uniqueNETADefinition(sp);
+            if (!neta.has_value())
+                return Messenger::error("Couldn't generate molecular partitioning for CIF - no unique NETA definition for the "
+                                        "fragmentIndices {} could be determined.\n",
```
Comment (jws-1) [inline src/io/import/cif.cpp]:
```suggestion
                                        "fragment {} could be determined.\n",
```

File: src/io/import/cif.cpp [hunk 7]
```diff
@@ -577,61 +577,86 @@ bool CIFHandler::detectMolecules()
...
+                                        EmpiricalFormula::formula(sp->atoms(), [](const auto &i) { return i.Z(); }));
```
Comment (jws-1) [inline src/io/import/cif.cpp]:
```suggestion
                                        sp->name());
```

File: src/io/import/cif.cpp [hunk 8]
```diff
@@ -577,61 +577,86 @@ bool CIFHandler::detectMolecules()
...
+
+            // Find instances of this fragmentIndices
+            instances = getSpeciesInstances(sp, *neta);
+
+            // Remove the current fragmentIndices and all copies
```
Comment (jws-1) [inline src/io/import/cif.cpp]:
```suggestion
            // Remove the current fragment and all copies
```

**Review by jws-1 — APPROVED** [review]
I really like this piece of work, particularly the introduction of the `CIFLocalMolecule` class. I think this cleans things up nicely. Some very minor comments, which may be simply my misunderstanding.

File: src/io/import/cif.cpp (hunk 2 above)
Comment (trisyoungs) [reply src/io/import/cif.cpp]: Ah, yes. Lazy search/replacing on my part!

File: src/io/import/cif.h (hunk 1 above)
Comment (trisyoungs) [reply src/io/import/cif.h:143]: :shrug:

**Pull request merged**